pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1300,24 +1303,39 @@ func configToStatus(ctx *domainContext, config types.DomainConfig,
 		// map from i=1 to xvda, 2 to xvdb etc
 		xv := "xvd" + string(int('a')+i)
 		ds.Vdev = xv
-		locationDir := verifiedDirname + "/" + dc.ImageSha256
-		log.Debugf("configToStatus(%v) processing disk img %s for %s\n",
-			config.UUIDandVersion, locationDir, config.DisplayName)
-		location, err := locationFromDir(locationDir)
-		if err != nil {
-			return err
+
+		var location string
+		if status.IsContainer {
+			// Check if status.ContainerImageID has "sha512-" substring at the beginning
+			if strings.Index(status.ContainerImageID, "sha512-") != 0 {
+				return fmt.Errorf("status.ContainerImageID should start with sha512-, but is %s", status.ContainerImageID)
+			}
+			//location = persistRktDataDir + "/cas/blob/sha512/" + string(status.ContainerImageID[7:9]) + "/" + status.ContainerImageID
+			location = filepath.Join(persistRktDataDir, "cas", "blob", "sha512", string(status.ContainerImageID[7:9]), status.ContainerImageID)
+		} else {
+			locationDir := verifiedDirname + "/" + dc.ImageSha256
+			log.Debugf("configToStatus(%v) processing disk img %s for %s\n",
+				config.UUIDandVersion, locationDir, config.DisplayName)
+
+			var err error
+			location, err = locationFromDir(locationDir)
+			if err != nil {
+				return err
+			}
 		}
 		ds.FileLocation = location
 		target := location
-		if !dc.ReadOnly {
-			// Pick new location for a per-guest copy
-			// Use App UUID to make sure name is the same even
-			// after adds and deletes of instances and device reboots
-			dstFilename := fmt.Sprintf("%s/%s-%s.%s",
-				rwImgDirname, dc.ImageSha256,
-				config.UUIDandVersion.UUID.String(),
-				dc.Format)
-			target = dstFilename
+		if !status.IsContainer {
+			if !dc.ReadOnly {
```

Comment:
Shouldn't ReadOnly always be set for container images?

Reply:
Currently Both ReadOnly and Persist are being passed as "False" to Domain manager.
This needs to be revisited.